integration/ledger/snapshot_test.go

```diff
@@ -93,184 +71,398 @@ var _ bool = Describe("Snapshot Generation and Bootstrap", func() {
+			couchProcess = append(couchProcess, couchProc)
+
+			By("installing chaincode on peer org2peer1")
+			nwo.InstallChaincode(setup.network, newlifecycleChaincode, org2peer1)
+
+			By("joining peer org2peer1 to the channel by snapshot")
+			joinBySnapshot(setup.network, setup.orderer, org2peer1, testchannelID, snapshotDir2, blockNum)
+
+			By("waiting for pvtdata to be reconciled on org2peer1")
+			waitForMarblePvtdataReconciliation(setup.network, org2peer1, testchannelID, newlifecycleChaincode.Name, "marble2")
+
+			// verify pvtdata reconciliation after joinbysnapshot
+			By("verifying marble2 pvtdata reconciliation on org2peer1")
+			assertPvtdataPresencePerCollectionConfig1(setup.network, newlifecycleChaincode.Name, "marble2", org2peer1)
+			By("verifying the new ownership of marble2")
+			marblechaincodeutil.AssertOwnershipInCollectionM(setup.network, testchannelID, newlifecycleChaincode.Name, "marble2", "jerry", org2peer1)
+
+			By("verifying marble1 does not exist")
+			marblechaincodeutil.AssertDoesNotExistInCollectionM(setup.network, testchannelID, newlifecycleChaincode.Name, "marble1", org2peer1)
+			marblechaincodeutil.AssertDoesNotExistInCollectionMPD(setup.network, testchannelID, newlifecycleChaincode.Name, "marble1", org2peer1)
+
+			// test 3: submit a request to generate snapshot again on a peer (org2peer1) bootstrapped from a snapshot
+			blockNumForNextSnapshot := nwo.GetLedgerHeight(setup.network, org2peer1, testchannelID)
+			By(fmt.Sprintf("generating a snapshot at blockNum %d on org2peer1 that was bootstrapped by a snapshot", blockNumForNextSnapshot))
+			submitSnapshotRequest(setup.network, testchannelID, blockNumForNextSnapshot, org2peer1, false, "Snapshot request submitted successfully")
+
+			// block for marble3 tx is in snapshot, but block for marble4 tx is post snapshot
+			By("adding marble3")
+			marblechaincodeutil.AddMarble(setup.network, setup.orderer, testchannelID, newlifecycleChaincode.Name,
+				`{"name":"marble3", "color":"blue", "size":35, "owner":"tom", "price":99}`, org2peer1)
+			assertPvtdataPresencePerCollectionConfig1(setup.network, newlifecycleChaincode.Name, "marble3")
+
+			By("adding marble4")
+			marblechaincodeutil.AddMarble(setup.network, setup.orderer, testchannelID, newlifecycleChaincode.Name,
+				`{"name":"marble4", "color":"blue", "size":35, "owner":"tom", "price":99}`, org2peer1)
+			assertPvtdataPresencePerCollectionConfig1(setup.network, newlifecycleChaincode.Name, "marble4")
+
+			By("verifying snapshot completed on org2peer1")
+			verifyNoPendingSnapshotRequest(setup.network, org2peer1, testchannelID)
+			snapshotDir := filepath.Join(setup.network.PeerDir(org2peer1), "filesystem", "snapshots", "completed", testchannelID, strconv.Itoa(blockNumForNextSnapshot))
+
+			// stop all the peers and only restart org2peer1
+			setup.stopPeers()
+			setup.startPeer(org2peer1)
+			setup.peers = []*nwo.Peer{org2peer1}
+			setup.network.Peers = setup.peers
+
```

Comment:
Not related to this PR, but we can do some refactoring (in a separate PR) of the `setup`, so that the state changed by the operations start and stop is handled internally.